src/main/java/be/seeseemelk/mockbukkit/block/data/WallSignMock.java

```diff
@@ -0,0 +1,56 @@
+package be.seeseemelk.mockbukkit.block.data;
+
+import java.util.Set;
+
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.type.WallSign;
+import org.jetbrains.annotations.NotNull;
+
+import static be.seeseemelk.mockbukkit.block.data.BlockDataKey.FACING;
+import static be.seeseemelk.mockbukkit.block.data.BlockDataKey.WATERLOGGED;
+
+public class WallSignMock extends BlockDataMock implements WallSign {
+
+	/**
+	 * Constructs a new {@link WallSignMock} for the provided {@link Material}. Only
+	 * supports materials in {@link Tag#WALL_SIGNS}
+	 * 
+	 * @param type The material this data is for.
+	 */
+	WallSignMock(@NotNull Material type) {
+		super(type);
+		checkType(type, Tag.WALL_SIGNS);
+		this.setFacing(BlockFace.NORTH);
+		this.setWaterlogged(false);
+	}
+
+	@Override
+	public @NotNull BlockFace getFacing() {
+		return super.get(FACING);
+	}
+
+	@Override
+	public void setFacing(@NotNull BlockFace facing) {
```

Comment:
The parameter's nullity should be checked. A test will also have to be added for this.
```suggestion
	public void setFacing(@NotNull BlockFace facing) {
		Preconditions.checkNotNull(facing, "BlockFace cannot be null");